tests/ProductionConfigTests.js

```diff
@@ -24,28 +24,710 @@ https://github.com/GPII/universal/blob/master/LICENSE.txt
 "use strict";
 
 var fluid = require("infusion"),
-    gpii = fluid.registerNamespace("gpii");
+    jqUnit = fluid.require("node-jqunit", require, "jqUnit"),
+    gpii = fluid.registerNamespace("gpii"),
+    kettle = fluid.registerNamespace("kettle");
 
 fluid.registerNamespace("gpii.tests.productionConfigTesting");
 
 fluid.require("%gpii-universal");
 
+fluid.logObjectRenderChars = 1024000;
+
 /*
- * ==============================================================================
- * Testing of untrusted local flow manager with the live cloud based flow manager
- * ==============================================================================
+ * ========================================================================
+ * Testing of untrusted local config with the live cloud based flow manager
+ * ========================================================================
  */
 
 require("./shared/DevelopmentTestDefs.js");
+require("./shared/FlowManagerSettingsGetTestDefs.js");
+require("./shared/FlowManagerSettingsPutTestDefs.js");
 
 gpii.loadTestingSupport();
 
-gpii.test.bootstrapServer(fluid.transform(gpii.tests.development.testDefs, function (testDefIn) {
+gpii.tests.productionConfigTesting.config = {
+    configName: "gpii.tests.untrusted.production.config",
+    configPath: "%gpii-universal/tests/configs"
+};
+
+gpii.tests.productionConfigTesting.settingsUserKey = {
```

Comment:
Can we use convertPrefs.js to cherry pick keys and prefs safes that are required for the production config tests so that we only need to maintain these data at testData/preferences directory.

Reply:
Possibly, but there are issues.  First, these production tests are run from the `vagrantCloudBasedContainers.sh` script,.  The preferences in testData/preferences have already been converted and loaded into the data base as snapsets by then.  They can't be used as 'user' prefs safes.

Another issue is how fixed the keys/safes in testData/preferences are.  By defining test keys and a 'user' prefsSafe here, we have complete control over what is put in the database (and later removed), and, hence, what to expect as a response to a request.  If we used something from testData/preferences what guarantee that those files won't change, and some test(s)' expected results will have to be updated?

Thirdly, at the moment, the test 'user' preference set (`os_gnome`) was constructed such that it could also be used in both production and development configurations.  In fact, it's what is used now in the development configuration.  These preferences for development testing do not exist in testData/preferences.  Do we want to add a set for `os_gnome` to the testData/preferences?  If so, they will be added as a snapset.

I'll think about it some more, but it's seems as if it's more trouble than it's worth.

Reply:
I agree that we don't wanna `os_gnome` into testData/preferences. What do you think about this solution:

1. Create tests/data/preferencesProductionTest directory to hold "user" prefs data that will be converted and loaded for production tests;
2. Add a one more step in the [postinstall step](https://github.com/GPII/universal/blob/master/package.json#L80) to convert this data to build/tests/dbDataForProduction where production config tests will pick up user data from.

Reply:
It comes to me later that the idea above of copying `os_gnome` and other prefs files for production config tests to a new tests/data/preferencesProductionTest directory is still duplicating these files, which also increases maintenance burden and is not ideal. The ideal approach is to have one single file used for various cases.

So a thought is to slightly improve ./scripts/convertPrefs.js to make it able to accept either a directory or an array of json files for conversion. This will allow preferences files from different source directories to be converted to single output files. And the step 1 above won't need to duplicate prefs files and create a new source directory.

Let me know what you think. Thanks.

Reply:
As we discussed yesterday, I've changed the code to load the keys and prefs safe from a common json files.

Reply:
Thanks, @klown. It looks great.